keps/sig-node/20200310-ensure-secret-pulled-images.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Ensure Secret Pulled Images
+authors:
+  - "@mikebrow"
+owning-sig: sig-node
+participating-sigs:
+  - sig-node
+reviewers:
+  - "@Random-Liu"
+  - "@yujuhong"
+approvers:
+  - "@dchen1107"
+editor: N/A
+creation-date: 2020-03-10
+last-updated: 2020-03-10
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# Ensure Secret Pulled Images
+
+## Table of Contents
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Examples](#examples)
+      - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+      - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives [optional]](#alternatives-optional)
+- [Infrastructure Needed [optional]](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+We will add support for ensuring images pulled with pod imagePullSecrets are
+always authenticated even if cached. We will add a new boolean field
+`ensureSecretPulledImages` to the pod spec. The default to false
```

Comment:
we shouldn't need a flag to do this in the pod spec.

this should be a security posture of the kubelet configuration and not the pod specification.

Reply:
updated .. should be resolved.